### PR TITLE
fix current-clamp generation

### LINKED ISSUE #1486: Cable Cell I/O does not load I_Clamp
Loading the attached .acc fails with
```
libc++abi.dylib: terminating with uncaught exception of type arborio::cableio_parse_error: Expression is not integer, real expression of the form (op <args>) nor tuple of the form (e0 e1 ... en) at :128:15
```
The issue seems to be that the time field in the envelope is not an integer.
```lisp
        (current-clamp 
          (envelope 
            ((0.000000 100.000000)))
          100.000000 2.323000))
```
The file was produced by
```c++
  auto cell = arb::cable_cell(builder.morph, builder.labels, decor);
  arborio::write_component(fd, cell);
```
and loaded as
```c++
  auto thing = arborio::parse_component(fd);
```

Here the critical file
```lisp
(arbor-component 
  (meta-data 
    (version "0.1-dev"))
  (cable-cell 
    (morphology 
      (branch 0 -1 
        (segment 0 
          (point 0.000000 0.000000 0.000000 10.000000)
          (point 20.000000 0.000000 0.000000 10.000000)
          0)
        (segment 1 
          (point 20.000000 0.000000 0.000000 4.000000)
          (point 60.000000 0.000000 0.000000 4.000000)
          0)
        (segment 2 
          (point 60.000000 0.000000 0.000000 2.750000)
          (point 100.000000 0.000000 0.000000 2.750000)
          0)
        (segment 3 
          (point 100.000000 0.000000 0.000000 2.333333)
          (point 140.000000 0.000000 0.000000 2.333333)
          0)
        (segment 4 
          (point 140.000000 0.000000 0.000000 2.125000)
          (point 180.000000 0.000000 0.000000 2.125000)
          0)
        (segment 5 
          (point 180.000000 0.000000 0.000000 2.000000)
          (point 220.000000 0.000000 0.000000 2.000000)
          0)
        (segment 6 
          (point 220.000000 0.000000 0.000000 1.916667)
          (point 260.000000 0.000000 0.000000 1.916667)
          0)
        (segment 7 
          (point 260.000000 0.000000 0.000000 1.857143)
          (point 300.000000 0.000000 0.000000 1.857143)
          0)
        (segment 8 
          (point 300.000000 0.000000 0.000000 1.812500)
          (point 340.000000 0.000000 0.000000 1.812500)
          0)
        (segment 9 
          (point 340.000000 0.000000 0.000000 1.777778)
          (point 380.000000 0.000000 0.000000 1.777778)
          0)
        (segment 10 
          (point 380.000000 0.000000 0.000000 1.750000)
          (point 420.000000 0.000000 0.000000 1.750000)
          0)))
    (label-dict 
      (region-def "all" 
        (join 
          (join 
            (join 
              (join 
                (join 
                  (join 
                    (join 
                      (join 
                        (join 
                          (join 
                            (join nil 
                              (join nil 
                                (segment 0)))
                            (join nil 
                              (segment 1)))
                          (join nil 
                            (segment 2)))
                        (join nil 
                          (segment 3)))
                      (join nil 
                        (segment 4)))
                    (join nil 
                      (segment 5)))
                  (join nil 
                    (segment 6)))
                (join nil 
                  (segment 7)))
              (join nil 
                (segment 8)))
            (join nil 
              (segment 9)))
          (join nil 
            (segment 10))))
      (region-def "soma_group" 
        (join nil 
          (join nil 
            (segment 0))))
      (region-def "dendrite_group" 
        (join 
          (join 
            (join 
              (join 
                (join 
                  (join 
                    (join 
                      (join 
                        (join 
                          (join nil 
                            (join nil 
                              (segment 1)))
                          (join nil 
                            (segment 2)))
                        (join nil 
                          (segment 3)))
                      (join nil 
                        (segment 4)))
                    (join nil 
                      (segment 5)))
                  (join nil 
                    (segment 6)))
                (join nil 
                  (segment 7)))
              (join nil 
                (segment 8)))
            (join nil 
              (segment 9)))
          (join nil 
            (segment 10))))
      (locset-def "Soma" 
        (root)))
    (decor 
      (place 
        (locset "Soma")
        (current-clamp 
          (envelope 
            ((0.000000 100.000000)))
          100.000000 2.323000))
      (place 
        (locset "Soma")
        (threshold-detector 0.000000)))))
```

### PR DESCRIPTION
**Pull request by noraabiakar**
Fixes incorrect `current-clamp` generated s-expression. 
Adds unit test.

Fixes #1486

### CODE REVIEW

**Comment by noraabiakar** [comment]
Nope, that was a mistake. Fixed now.

**Review by thorstenhater — APPROVED** [review]
👍

File: arborio/cableio.cpp [hunk 1]
```diff
@@ -60,7 +60,7 @@ s_expr mksexp(const i_clamp& c) {
     std::vector<s_expr> evlps;
     std::transform(c.envelope.begin(), c.envelope.end(), std::back_inserter(evlps),
         [](const auto& x){return slist(x.t, x.amplitude);});
-    auto envelope = slist("envelope"_symbol, slist_range(evlps));
+    auto envelope = s_expr{"envelope"_symbol, slist_range(evlps)};
     return slist("current-clamp"_symbol, envelope, c.frequency, c.phase);
```
Comment (bcumming) [inline arborio/cableio.cpp:64]: A small optimization using a move: `return slist("current-clamp"_symbol, std::move(envelope), c.frequency, c.phase)`

**Comment by thorstenhater** [comment]
bors try

File: arborio/cableio.cpp (hunk 1 above)
Comment (noraabiakar) [reply arborio/cableio.cpp:64]: Okay!

**Comment by thorstenhater** [comment]
bors try

**Pull request merged**